nemo/collections/nlp/data/language_modeling/megatron/t5_speechlm_dataset.py

```diff
@@ -312,9 +335,9 @@ def __getitem__(self, idx):
         context_tokens, context_tokens_len = self.list_to_tensor(context_tokens)
         question_tokens, question_tokens_len = self.list_to_tensor(question_tokens)
 
-        if doc["question_type"] != "SPEECH" and doc["context_type"] == "SPEECH":
+        if doc["question_type"] == "TEXT" and doc["context_type"] != "TEXT":
             question_tokens = pad_text_to_speech_dims(question_tokens, self.tokenizer.pad_id)
-        if doc["context_type"] != "SPEECH" and doc["question_type"] == "SPEECH":
+        if doc["context_type"] == "TEXT" and doc["question_type"] != "TEXT":
```

Comment:
any reason to use "TEXT" instead of "SPEECH" ?

Reply:
Yes, because now in the multitask dataset, we have other context_types like "AUDIOCODEC" (in which case context filepath is the path to the encodec codes .pt file instead of the .wav file in Speech), which also needs 8 codebooks. So to handle both audiocodec and speech in the same way, I am using != TEXT